conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp

```diff
@@ -680,16 +684,35 @@ TEST_F(
                             [&] { return t_params.get_process_state_flag; });
 
     uint32_t engine_type = 0;
-    while (engine_type != ZES_ENGINE_TYPE_FLAG_COMPUTE) {
-      auto processes = lzt::get_processes_state(device, count);
+    ze_result_t result{};
+    std::chrono::time_point<std::chrono::system_clock> start =
+        std::chrono::system_clock::now();
+    std::chrono::duration<double> elapsed_seconds = std::chrono::seconds{0};
+
+    while (engine_type != ZES_ENGINE_TYPE_FLAG_COMPUTE &&
+           elapsed_seconds < std::chrono::seconds{30}) {
+      auto processes = lzt::get_processes_state(device, count, result);
+      EXPECT_EQ(ZE_RESULT_SUCCESS, result);
+      if (result == ZE_RESULT_ERROR_UNSUPPORTED_FEATURE) {
+        LOG_INFO << "zesDeviceProcessesGetState is Unsupported on this device!";
+        break;
+      }
       for (auto process : processes) {
         engine_type = process.engines;
       }
       processes.clear();
+      elapsed_seconds = std::chrono::system_clock::now() - start;
     }
 
     thread.join();
-    EXPECT_EQ(engine_type, ZES_ENGINE_TYPE_FLAG_COMPUTE);
+    if (elapsed_seconds >= std::chrono::seconds{30}) {
+      FAIL() << "TimeOut of 30 seconds elapsed while waiting for engine type "
```

Comment:
The Fail message is not clear to me. But if its correct please ignore my comment